sound/soc/sof/topology.c

```diff
@@ -1258,6 +1261,13 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 		goto err;
 	}
 
+	/* set default max volume values to control */
+	cdata = scontrol->control_data;
+	for (i = 0; i < scontrol->num_channels; i++) {
+		cdata->chanv[i].channel = i;
+		cdata->chanv[i].value = VOL_ZERO_DB;
+	}
+
```

Comment:
do you happen to know how this will work or interfere with alsactrl trying to restore previous values?

Reply:
it should work, because this is topology loading, and I guess any alsactl command should not come in the middle (?). Not sure what happens if control is called in the middle of topology loading. PM is caching the stuff and in reload this part is never called. However, good point, I'll re-enable my alsactl and give it a try anyway.

Reply:
@plbossart I tested with following scenario in my ubuntu:
1) alsactl removed, so "first boot" set both pga channels to 0db
2) I changed manually from alsamixer the pga volumes and re-enabled alsactl
3) after boot the pga volumes are still in positions I set in 2

So I can't still guarantee fully all use cases, but this basic one is working.

Reply:
works for me as well, I took the code and all the settings were moved to 0dB on all PGAs. I changed one of the gains, rebooted and the values were kept, so it looks functional to me. Will merge, thanks Jaska!